File: lua/day6_part1_2021.lua

local file = io.open("input.txt", "r")
if not file then
    print("Error opening file")
    return
end

local input = file:read("*line")
file:close()

local fishes = {0, 0, 0, 0, 0, 0, 0, 0, 0}
for fishStr in input:gmatch("%d+") do
    local fish = tonumber(fishStr)
    fishes[fish + 1] = fishes[fish + 1] + 1
end

for day = 1, 80 do
    local newFish = fishes[1]
    for i = 1, 8 do
        fishes[i] = fishes[i + 1]
    end
    fishes[7] = fishes[7] + newFish
    fishes[9] = newFish
end

local totalFish = 0
for _, fish in ipairs(fishes) do
    totalFish = totalFish + fish
end

print(totalFish)

File: go/day6_part1_2021.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close()

	var fishes [9]int
	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanWords)
	scanner.Scan()
	input := scanner.Text()
	fishStrs := strings.Split(input, ",")
	for _, fishStr := range fishStrs {
		fish, _ := strconv.Atoi(fishStr)
		fishes[fish]++
	}

	for day := 1; day <= 80; day++ {
		newFish := fishes[0]
		for i := 1; i < len(fishes); i++ {
			fishes[i-1] = fishes[i]
		}
		fishes[6] += newFish
		fishes[8] = newFish
	}

	totalFish := 0
	for _, fish := range fishes {
		totalFish += fish
	}

	fmt.Println(totalFish)
}